alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderGlobalUIConfig.java

```diff
@@ -58,4 +74,29 @@ public ProviderKey getProviderKey() {
         return providerKey;
     }
 
+    private Collection<String> validateDuplicateNames(FieldValueModel fieldToValidate, FieldModel fieldModel) {
+        try {
+            List<ConfigurationModel> configurations = configurationAccessor.getConfigurationsByDescriptorType(DescriptorType.PROVIDER);
+            if (!configurations.isEmpty()) {
+                List<ConfigurationModel> modelsWithName = configurations.stream()
+                                                              .filter(configurationModel ->
+                                                                          configurationModel.getField(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME)
+                                                                              .flatMap(ConfigurationFieldModel::getFieldValue)
+                                                                              .filter(configName -> configName.equals(fieldToValidate.getValue().orElse("")))
+                                                                              .isPresent())
+                                                              .collect(Collectors.toList());
+                if (modelsWithName.size() > 1) {
+                    return List.of(ERROR_DUPLICATE_PROVIDER_NAME);
```

Comment:
Should this be a database constraint? I don't think this is the appropriate place to handle this error.

Reply:
We could move this duplicate validation into the database accessors.

Reply:
A database exception won't get associated as a field exception and then we can't display the error of a duplicate name with the name text field in the form.  We perform the same validation with job names and construct the the field exception in the JobConfigActions.  If I throw a database exception it won't be tied to a field.  We should add a constraint to the database that the configuration name be unique

Reply:
My bad, I realize this is the validator, not the endpoint.